### ISSUE TITLE
class "Component" is inherited and used by Widgets only and it has an inconsistent description

### ISSUE THREAD

**Issue by Zughy** [LibreSprite/LibreSprite#152]
As said already on Matrix/Discord, I was trying to better understand the code and I notice this in `Component.h`.
![imagen](https://user-images.githubusercontent.com/63455151/130532323-0b7f0893-b9eb-4cc3-8c42-c58ac7e9706a.png)
But this isn't 100% true, because menus are... well, widgets.
![imagen](https://user-images.githubusercontent.com/63455151/130532380-ac7496d8-094d-4192-9cbf-60846769e3e0.png)

Widget is the only class extending Component, and `ui.h` the only other file including its header. However, if commented out, everything seems to work fine anyway.

In short, there is an abstract class which might be unnecessary (I'm saying *might* because it needs further investigation). I've also checked Aseprite code and the situation haven't changed in 5 years. No one has ever pointed that out, which honestly surprises me. So either it's just a matter of wording or, worst case scenario, the hierarchy needs some rework

**Comment by maujin111** [comment]
In my experience, sometimes if you delete or change something in the code that seems unnecessary, something stops working well. If it is not broke, do not fix it. Maybe in the future this will be changed, but for now I think it is better to leave it as is
